packages/lodestar/src/network/gossip/validation/aggregateAndProof.ts

```diff
@@ -102,59 +92,67 @@ export async function validateAggregateAttestation(
-    )
-  ) {
-    logger.warn("Rejected gossip aggregate and proof", {reason: "invalid selection proof signature", ...logContext});
-    return ExtendedValidatorResult.reject;
-  }
 
-  if (
-    !isValidAggregateAndProofSignature(
-      config,
-      state,
-      computeEpochAtSlot(config, attestation.data.slot),
-      aggregator,
-      aggregateAndProof
-    )
-  ) {
-    logger.warn("Rejected gossip aggregate and proof", {reason: "invalid signature", ...logContext});
-    return ExtendedValidatorResult.reject;
-  }
+  try {
+    const committee = epochCtx.getBeaconCommittee(attestation.data.slot, attestation.data.index);
+    if (!committee.includes(aggregateAndProof.message.aggregatorIndex)) {
+      logger.warn("Rejected gossip aggregate and proof", {reason: "aggregator not in committee", ...logContext});
+      return ExtendedValidatorResult.reject;
+    }
+    if (!isAggregatorFromCommitteeLength(config, committee.length, aggregateAndProof.message.selectionProof)) {
+      logger.warn("Rejected gossip aggregate and proof", {reason: "not valid aggregator", ...logContext});
+      return ExtendedValidatorResult.reject;
+    }
+    const aggregator = epochCtx.index2pubkey[aggregateAndProof.message.aggregatorIndex];
+    if (
+      !isValidSelectionProofSignature(
+        config,
+        state,
+        attestation.data.slot,
+        aggregator,
+        Signature.fromCompressedBytes(aggregateAndProof.message.selectionProof.valueOf() as Uint8Array)
+      )
+    ) {
+      logger.warn("Rejected gossip aggregate and proof", {reason: "invalid selection proof signature", ...logContext});
+      return ExtendedValidatorResult.reject;
+    }
+
+    if (
+      !isValidAggregateAndProofSignature(
+        config,
+        state,
+        computeEpochAtSlot(config, attestation.data.slot),
+        aggregator,
+        aggregateAndProof
+      )
+    ) {
+      logger.warn("Rejected gossip aggregate and proof", {reason: "invalid signature", ...logContext});
+      return ExtendedValidatorResult.reject;
+    }
 
-  //TODO: once we have pool, check if aggregate block is seen and has target as ancestor
+    // TODO: once we have pool, check if aggregate block is seen and has target as ancestor
 
-  if (!isValidIndexedAttestation(epochCtx, state, epochCtx.getIndexedAttestation(attestation))) {
-    logger.warn("Rejected gossip aggregate and proof", {reason: "invalid indexed attestation", ...logContext});
+    if (!isValidIndexedAttestation(epochCtx, state, epochCtx.getIndexedAttestation(attestation))) {
+      logger.warn("Rejected gossip aggregate and proof", {reason: "invalid indexed attestation", ...logContext});
+      return ExtendedValidatorResult.reject;
+    }
+  } catch (e) {
+    // if any errors are thrown in the process of checking for REJECTs, return a REJECT
+    logger.warn("Rejected gossip aggregate and proof", {reason: e.message, ...logContext});
```

Comment:
can you just change that in GossipMessageValidator.isValidIncomingAggregateAndProof to be reject as we already catch errors there

Reply:
yeah i was wondering if there was any reason why the ignore was thrown in that particular try/catch.  do you happen to know?  or is it mostly just an arbitrary default value?  if we change that try/catch to throw a reject, then there could be a possibility that while we're computing some values for a case that wants to throw an ignore that we would run into the same problem by throwing a reject by default.

Reply:
i will probably also be restructuring the way the errors are thrown once we get the error type that @wemeetagain  said he was going to be making a PR for, so this might not apply anymore when that happens.

Reply:
> yeah i was wondering if there was any reason why the ignore was thrown in that particular try/catch. do you happen to know?

No particular reason, I've added those top level try catch in case we have some bug or encounter some unexpected value. My reasoning for ignore instead of reject was that if we have bug it's better not to censor messages. But you are probably right, it's better to reject in those situations.

Reply:
Well, I think it depends on where the bug would be happening.  If the bug happens while trying to check for a condition that should throw an ignore if the condition is not met, then we should throw ignore, and if the bug happens when we are checking for a condition that should throw a reject if the condition is not met, then we should throw a reject.  I think maybe to keep this PR minimal and specific to spec-compliance, I will make a separate PR for restructuring the errors and incorporating the work done in #1579

Reply:
Though I'm not 100% if my assumption is correct that we should throw a reject or ignore depending on whether the bug happens in a reject or ignore condition check.  Open to questioning that assumption, if there's a good case for it.
On one hand, we could have a condition where we intentionally throw an ignore or reject because some data is missing from the gossip, but on the other hand, we may also have a bug that is something wrong with Lodestar, so maybe in that case we want a generic ignore or reject for catching these bugs.  I think a generic ignore for catching bugs could be good since it doesn't punish the user like reject would, but I'm not sure.

For example, in the case of this error here (https://github.com/ChainSafe/lodestar/blob/7249507d26601f8c14864ace476f6808601e4da9/packages/lodestar-beacon-state-transition/src/fast/util/epochContext.ts#L133) that would get thrown while retrieving the beacon committee to see if the aggregator is in the committee (the case brought up in the description of this PR), would this error only happen if there was a bug in the code?  Or could this also happen because of some bad data in the incoming gossip?

Reply:
To me, it seems like a reject should be returned in that specific case, but I'm not sure about all cases.